stackit/internal/services/ske/kubeconfig/resource.go

```diff
@@ -0,0 +1,343 @@
+	_ resource.ResourceWithConfigure  = &kubeconfigResource{}
+	_ resource.ResourceWithModifyPlan = &kubeconfigResource{}
+)
+
+type Model struct {
+	Id          types.String `tfsdk:"id"` // needed by TF
+	ClusterName types.String `tfsdk:"cluster_name"`
+	ProjectId   types.String `tfsdk:"project_id"`
+	Kubeconfig  types.String `tfsdk:"kube_config"`
+	Expiration  types.Int64  `tfsdk:"expiration"`
+	Refresh     types.Bool   `tfsdk:"refresh"`
+	ExpiresAt   types.String `tfsdk:"expires_at"`
+}
+
+// NewKubeconfigResource is a helper function to simplify the provider implementation.
+func NewKubeconfigResource() resource.Resource {
+	return &kubeconfigResource{}
+}
+
+// kubeconfigResource is the resource implementation.
+type kubeconfigResource struct {
+	client *ske.APIClient
+}
+
+// Metadata returns the resource type name.
+func (r *kubeconfigResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_ske_kubeconfig"
+}
+
+// Configure adds the provider configured client to the resource.
+func (r *kubeconfigResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerData, ok := req.ProviderData.(core.ProviderData)
+	if !ok {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Expected configure type stackit.ProviderData, got %T", req.ProviderData))
+		return
+	}
+
+	var apiClient *ske.APIClient
+	var err error
+	if providerData.SKECustomEndpoint != "" {
+		apiClient, err = ske.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithEndpoint(providerData.SKECustomEndpoint),
+		)
+	} else {
+		apiClient, err = ske.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithRegion(providerData.Region),
+		)
+	}
+
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Configuring client: %v. This is an error related to the provider configuration, not to the resource configuration", err))
+		return
+	}
+
+	r.client = apiClient
+	tflog.Info(ctx, "SKE kubeconfig client configured")
+}
+
+// Schema defines the schema for the resource.
+func (r *kubeconfigResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	descriptions := map[string]string{
+		"main":         "SKE kubeconfig resource schema. Must have a `region` specified in the provider configuration.",
+		"id":           "Terraform's internal resource ID. It is structured as \"`project_id`,`cluster_name`,`uuid`\".",
+		"cluster_name": "Name of the SKE cluster.",
+		"project_id":   "STACKIT project ID to which the cluster is associated.",
+		"kube_config":  "Raw kube config.",
+		"expiration":   "Expiration time of the kubeconfig, in seconds. The default is 3600s (1h).",
```

Comment:
I decided to keep it with this name for consistency with the API where it is called `expirationSeconds`. Just removed the `seconds` part because I think it doesnt make sense and it's better to have in the description.

In the terraform we try to keep closer to the API